### DISCUSSION TITLE
Prevent EntityDialog or Panel from Opening when a condition is not met

### DISCUSSION THREAD

**Discussion by flashyjunior** [serenity-is/Serenity#7304] [Help] [answered]
Hi,
How do I prevent a new dialog from opening when a user clicks on the Add button on the Grid page.
I check if a period has been opened to allow a new entry, if its not then the dialog should not open at all when the add button is clicked
I tried the below in the dialog.tsx but it didnt work
```
onOpen(handler,opt?) {
    super.onOpen(handler,opt.before);

    var userdata = getRemoteData('UserData');
    var leaveperiodid = toId(userdata["LeavePeriodId"])

    if (leaveperiodid != 0) {
        alertDialog("No Leave Period has been opened for this year... Please contact the administrator");
        handler.preventDefault()
    }

}

```

**Comment by arunkumar12712** [comment, marked as answer]
try this

protected afterLoadEntity() {
    super.afterLoadEntity();
    
    this.onOpen((e) => {
        if (leaveperiodid != 0)
            e.preventDefault();
    }, {
        before: true
    });
}
